devito/ops/transformer.py

```diff
@@ -138,29 +141,29 @@ def create_ops_dat(f, name_to_ops_dat, block):
         padding = f.padding[:time_pos] + f.padding[time_pos + 1:]
         halo = f.halo[:time_pos] + f.halo[time_pos + 1:]
         d_p_val = tuple(sympify([p[0] + h[0] for p, h in zip(padding, halo)]))
```

Comment:
I **think** you wanna use `_size_halo` and in particular `_size_halo.left` and `_size_halo.right` instead of accessing `halo` and  `padding` directly, with explicit integer indexing (which is somewhat like having magic numbers in the code)

Reply:
I managed to replace the `halo` and `padding` variables with the `_size_halo` that you said. 

But, `_size_halo` also bring the time dimension halo (although it is 0 in my tests), and in this part we are only interested in the space dimensions, so the 3 lines above got replace by this:

`d_p_val = sympify(f._size_halo.left[time_pos+1:])` 

I didn't manage to get rid of the `sympify` at this point. When I delete it, I got the following error: 
`AttributeError: 'int' object has no attribute 'is_Indexed'` 

So we need the sympfy to transform the int into Sympy objects again.

Reply:
> But, _size_halo also bring the time dimension halo (although it is 0 in my tests), and in this part we are only interested in the space dimensions,

does
```
padding = [f._size_padding[d] for d in f.space_dimensions]
```
help?
Analogous for halo.
It should give you what you need while retaining simplicity and clarity